### PR TITLE
components: Promote VStack and HStack

### PR DESCRIPTION
**Pull request by sarayourfriend**
## Description
Promote the HStack and VStack components to experimental exports from `@wordpress/components`. These are brand new components.

## How has this been tested?
Storybook still works. There's nothing to test otherwise because they're new components.

## Types of changes
New feature.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [N/A] My code follows the accessibility standards. 
- [N/A] I've tested my changes with keyboard and screen readers. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate. 
- [x] I've updated all React Native files affected by any refactorings/renamings in this PR (please manually search all `*.native.js` files for terms that need renaming or removal).

### CODE REVIEW

File: packages/components/CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,7 @@
 ### New Feature
 
 -   `Flex` and its companion components have been re-written from the ground up and the `isReversed` prop has been deprecated.
+-   Two new layout components, `HStack` and `VStack`, have been added under the experimental namespace.
```
Comment (gziolo) [inline packages/components/CHANGELOG.md]: Experimental features by design should not be documented. We should either ignore them here or expose as official APIs.

File: packages/components/src/h-stack/README.md [hunk 2]
```diff
@@ -1,13 +1,19 @@
...
 ## Usage
 
 `HStack` can render anything inside.
 
 ```jsx
-import { HStack, Text, View } from '@wordpress/components/ui';
+import {
+	__experimentalHStack as HStack,
+	__experimentalText as Text,
+	__experimentalView as View,
```
Comment (gziolo) [inline packages/components/src/h-stack/README.md:15]: I assume that `View` is going to be moved quite soon.

File: packages/components/src/h-stack/README.md [hunk 3]
```diff
@@ -83,7 +89,12 @@ Determines if children should wrap.
 When a `Spacer` is used within an `HStack`, the `Spacer` adaptively expands to take up the remaining space.
 
 ```jsx
-import { HStack, Spacer, Text, View } from '@wordpress/components/ui';
+import {
+	__experimentalHStack as HStack,
+	__experimentalSpacer as Spacer,
```
Comment (gziolo) [inline packages/components/src/h-stack/README.md:94]: There is no `Spacer` in `@wordpress/components` yet. Should we replace it with something else in this README file?

File: packages/components/src/v-stack/README.md [hunk 4]
```diff
@@ -83,6 +89,13 @@ Determines if children should wrap.
 When a `Spacer` is used within an `VStack`, the `Spacer` adaptively expands to take up the remaining space.
 
 ```jsx
+import {
+	__experimentalSpacer as Text,
```
Comment (gziolo) [inline packages/components/src/v-stack/README.md]:
```suggestion
	__experimentalSpacer as Spacer,
```

Or we replace this component with something else. In fact, we might want to remove the whole section for now.

File: packages/components/src/v-stack/README.md [hunk 5]
```diff
@@ -103,6 +116,13 @@ function Example() {
 `Spacer` can also be used in-between items to push them apart.
 
 ```jsx
+import {
+	__experimentalSpacer as Text,
```
Comment (gziolo) [inline packages/components/src/v-stack/README.md]:
```suggestion
	__experimentalSpacer as Spacer,
```

and same as above.

**Comment by gziolo** [comment]
There is no story for `VStack`.

A story for `HStack` looks this way:

Screen Shot 2021-04-30 at 13 47 48

**Review by gziolo — APPROVED** [review]
@sarayourfriend, I rebased this PR and applied all the necessary changes to make the story for `HStack` render properly in Storybook. I also applied a few changes to match with previously moved components.

The one remaining question is what do we do about all mentions of the `Spacer` component.

**Comment by sarayourfriend** [comment]
We can migrate Spacer as part of this PR? Or add it in a previous PR that this PR then is blocked by?

**Comment by sarayourfriend** [comment]
I opened a Spacer PR here: https://github.com/WordPress/gutenberg/pull/31509 We can merge it first and then incorporate it into the stories for HStack and VStack.

**Comment by gziolo** [comment]
> I opened a Spacer PR here: #31509 We can merge it first and then incorporate it into the stories for HStack and VStack.

There is no need to wait. I mostly raised it so we are aware that documentation contains `Spacer` component that doesn't exist yet and `View` component is inside the `/ui` subfolder.

**Pull request merged**